File: CodingPractise/leet_code/array_bruteforce_twosum.py
class Solution:
    def twoSum(self, nums, target):

        spare_list = {}

        for indexing,  itr in enumerate(nums):
            compliment = target - itr

            if compliment in spare_list:
                return [spare_list[compliment], indexing]

            spare_list[itr] = indexing

        return []




test = Solution()

nums = [3,4,5,6]

target = 8

res = test.twoSum(nums, target)

print(res)

File: CodingPractise/leet_code/array_bruteforce_twosum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

    int target_len = nums.length;

    Map<int, int> storage_map = new Map<int, int>();

    for(int itr=0; itr< target_len; itr++)
    {
        int complement = target - nums[itr];

        if( storage_map.containsvalue(complement) )
        {
            return [storage_map[complement] ,itr]];
        }



        storage_map.put(nums[itr], itr);

    }

    return {-1,-1};


}